tests/unittests/tests-crypto/tests-crypto-sha256.c

```diff
@@ -17,48 +18,75 @@
 
 #include "tests-crypto.h"
 
+static int compare_str_vs_digest(const char *str,
+                                 const unsigned char hash[SHA256_DIGEST_LENGTH])
+{
+    char ch[3] = { 0, 0, 0 };
+    size_t iter_hash = 0;
+
+    for (size_t i = 0; i < strlen(str); i += 2) {
+        ch[0] = str[i];
+        ch[1] = str[i + 1];
+        char *end;
```

Comment:
youre right, basically we can replace it completely by just passing `NULL` to `strtol()` instead of this pointer.
I will adjust it this afternoon.